### PR TITLE
Blood Vessel Lives Matter

### PR DESCRIPTION
**Pull request by DeityLink**
Frankly I don't think this should even be controversial, it's more for my immulsions that someone shouldn't refill their blood in two seconds from a bucket. At worse now you'll have to turn it into pills first.

TL;DR:
* bugfix: Non-vampires can no longer just drink blood from a cup and not get sick

detailed:
:cl:
* experiment: Changed the way blood is allowed to enter your vessels!
* experiment: The following items still allow blood to enter your vessels as usual. Syringes (and syringe guns, both normal and mecha), IV Drips, Pills, Blood Tomatoes, Hyposprays (and autoinjectors).
* experiment: Drinking blood any other way (such as from a glass or a bucket) will send it to the rest of your body instead, which isn't an issue in small quantities, but large quantities will eventually make you sick.
* experiment: This doesn't apply to Vampires! (more on that very soon)

### CODE REVIEW

**Comment by Irradiation** [comment]
I know it's a mix of your immulsions, only learning now that this was a feature and essentially a vampire-related PR that was rightfully told to be made on its own due to atomicity, but I kindly invite you to bugger the fuck off over a [FEATURE] that seldom gets used due to:
-Blood Tomatoes being generally non-existent.
-People even knowing that blood tomatoes refill your blood.
-People even knowing this feature is a thing.
-People in medbay having the time to bother making blood pills.
-People outside of medbay can hardly even slurp random blood due to the difficulty of its application (Impossibility to know a certain blood type outside of pre-emptively succing out your own blood for slurpies, which hardly replenish your blood vessels regardless.).

**>it's more for my immulsions that someone shouldn't refill their blood in two seconds from a bucket**

Remember that time you needed a valid blood type in order to slurp down blood or else it would turn instantly into toxins without replenishing your blood? I do. Meanwhile, short of your vampire changes being merged at #21707, this change will still make vampires sick for slurping an incompatible bloodtype.

**>eating a blood tomato gives you direct blood by essentially slurping it while eating
>but slurping from a bucket isn't OK**

Yeah okay, inb4 meme plant magic.


**>//it really shouldn't be intravenous but oh well. Set to FALSE once people get over it.**

Ah, the tried and true tactic of going at it until people get tired of fighting, the passing it when nobody cares anymore. I will admit that this is an attack on the ideology behind the comment and not the actual variable functionality, which moves into

**>At worse now you'll have to turn it into pills first.**

Which is essentially the only way this feature is ever used outside of the rarer blood tomatoes, but that's not necessarily a reason for removing the feature and essentially paving the way towards a complete removal. I understand that you've mentioned in the coder discord that you're keeping the main functionalities as players (including me) wish for the "meta" to stay that way, but we've been through enough medbay PR fuckery to know that it likely won't stop here.


**tl;dr You're removing a functionality that has bothered no one, requires preparation to work and is even in-lore justified via the entire "blood tomatoes replenish your blood by slurping blood tomato juice". Short of scrapping the entire PR, the vampire description tidbits are interdasting and the intravenous variable could be kept if admemes want to meme things.**

**Comment by Exxion** [comment]
So you're saying that because it never gets used, it shouldn't get removed?

**Comment by SGuivenne** [comment]
@Exxion thats a shockingly large amount of ss13 mechanics.

**Comment by Exxion** [comment]
Guess I should clarify
I'm not saying everything that's never used should be removed, but that "it's never used" doesn't make any sense as an argument against removing something
So that puts the tally at:
Remove: Immulsions
Keep: ???

**Comment by Irradiation** [comment]
It shouldn't get removed because the justification for its removal is based on a theorycrafted premise of somebody who just learned about it, an immulsion issue that ignores how the feature is legitimate in the first place and the invitation of a slippery slope that has removed other features in a similar fashion.

**Comment by Exxion** [comment]
If it's theorycrafting, nobody uses it
If nobody uses it, this affects nothing, so why do you care?

**Comment by Irradiation** [comment]
The theorycrafting bit was in regards to the assumption "that someone shouldn't refill their blood in two seconds from a bucket". This tidbit seldom (if at all) happens, but blood tomatoes and pills certainly happen and aren't quite as much theorycrafting-tier.

>If nobody uses it, this affects nothing, so why do you care?

Does this mean that we should remove largely unused features just because we can, they mildly bother our immulsions while already justified and because said feature is largely unused?

**Comment by DeityLink** [comment]
@Irradiation reguarding #21707 ignore it for now, there are un-pushed commits that I'm witholding until this PR gets sorted first.

**Comment by Irradiation** [comment]
Fair enough @DeityLink

**Comment by Exxion** [comment]
This doesn't remove blood tomatoes or blood pills.
Also, yes, if there's a reason to remove something and no reason to keep it, it seems obvious that it should be removed
And you keep saying it's "justified" but haven't actually said how or why yet

**Comment by DeityLink** [comment]
>Does this mean that we should remove largely unused features just because we can

When an alternative that makes sense and is already used by most players exists then yeah.

>they mildly bother our immulsions while already justified

highly debatable

>and because said feature is largely unused?

you're repeating yourself, and again yes.

**Comment by Irradiation** [comment]
At the cost of repeating myself once more, 

>eating a blood tomato gives you direct blood by essentially slurping it while eating
>but slurping from a bucket isn't OK

Which is to say, the mechanical logic behind blood tomatoes is that they contain the blood reagent and when you consume said blood tomato, you absorb the blood reagents in the same way that eating carrots will make you absorb eye-healing carrot juice. This is the "justification" behind it.

Coincidentally, if you would happen to splice blood tomatoes blood into another plant, wouldn't this mean that eating said other plant **would not** refill your blood, as only **blood **tomatoes** have the intravenous variable to allow blood chugging. Or maybe botany code is magical enough that this variable would be transferred to the different plant.

**Comment by Exxion** [comment]
That's a fair point, but on a detail of the PR rather than the PR as a whole

**Comment by Irradiation** [comment]
Err, edited my post to add more content to the answer,

**Comment by Exxion** [comment]
So how exactly is the reason illegitimate?

**Comment by Exxion** [comment]
Oh I didn't see the higher-up edits
The way that medicines work is entirely different from the way blood works, as evidenced by the fact that this PR is possible at all. This is on purpose, or else you would bleed out any chems you had ingested. Blood isn't on equal footing with other chemicals regardless of this PR, for good reason.

EDIT: I mean it would make sense for medicines to have to be in your bloodstream to work, but somehow I suspect you'd object to that change even more than this.

**Comment by DeityLink** [comment]
Last thing before I go to bed, as I said to Irrad via Discord, the ultimate goal is to clean up the oldcode weirdness so that we may expand upon blood and reagents without having to fear breaking stuff. I mean, if you like drinking blood so much we could add mutations or chemistry recipes that let you digest it.

I'm really tired though so I'll finish that PDA PR and go to bed.

**Comment by Irradiation** [comment]
We haven't had a good experience with trying to rework chemical intake and yeah, trying to be realistic with how chemicals in your bloodstream _is_ more realistic and could be fun were this server more realism-oriented, slow paced and not filled with mobs that 360 noscope break your legs, but it's one of those QoL things that spessmen ignore for the purpose of the gameplay we have on /vg/.

**Comment by SonixApache** [comment]
Hey, cool, it's the argument I had with the medbay crew during/after the whole waterpotass harmdrag thing.

Sure, this PR would make blood more 'realistic' and do good for your immurshuns but doesn't really make it any more fun or skillful.
Not to mention that it only makes the number one killer, bloodloss, an even stronger opponent.

**Comment by DeityLink** [comment]
The usual ways of fighting bloodloss that 99% of the player use and that were design to combat bloodloss still work though. And ultimately vampire players still (will) benefit from drinking blood (once #21707 gets in). What if I had just put in the changelog:
* bugfix: Non-vampires can no longer just drink blood from a cup and not get sick?

**Comment by DeityLink** [comment]
@Irradiation @SonixApache ok now hear me out.

What if, blood in your body, gave you blood back like iron does? This way you can still consume it to some extent, and it only gets ugly in large quantities just like alcohol.

**Comment by Kammerjunk** [comment]
what if fuck off

**Comment by SonixApache** [comment]
@DeityLink that should be a thing in the first place, all things considered.
But I stand by what I've said in the discord, which is;
 - Make this whole shebang only apply for vampires, since they have a weird unique physiology
 - Make succing live players more effective at the same time

**Comment by bathosbathtime** [comment]
This is removing a tool available to the players to deal with a problem often presented by the game for no reason other than immersions.  It is an awful tool with a near if not zero percent use compared to the alternative solutions available.
I disagree with removing it but it is an insignificant change.
The blood sickness, blood tomatoes and blood pills half-and-half realism is also upsetting me but I might just be a spastic.

**Comment by Kammerjunk** [comment]
What Bathos said and also copied from Discord from earlier ao it's somewhere it can actually be seen
>@Corentin "Deity" Pauvrasseau not logged into GitHub on my phone and I'm posting from work but the essence of the "why not remove" that I think Irradiation is trying to get at (but not sure because I only skimmed it) is that
A) "nobody uses" is - or at least was when I played - a gross understatement and
B) fun things you can learn to do are healthy for the game and learning that you can rather messily mess around with blood to make pills that restore less blood but faster is one such thing 

also I know the position that it's healthy for the game isn't supported in the self-quote but I'm about to go to sleep and I figure the logic is straight-forward

also optionally C) current medbay players not doing it isn't conclusive evidence for choosing not to do it
separated both because it wasn't in my original message and because it relies kn neither A nor B to be a (counter-)point 

edit: some poor lady named Mary Corention accidentally got that mention, sorry Mary

**Comment by bathosbathtime** [comment]
To address the suggestion of making consumed blood act like iron, I am going to ask why you would want to make an extremely situational and already awful way to transfuse blood even worse.
I am not saying that it should be buffed but making it a worse version of what it already is is simultaneously unrealistic and mechanically unsatisfying.

**Comment by DeityLink** [comment]
You're right, it's was a tentative compromise but I'll drop it.

**Comment by DeityLink** [comment]
Poll ended after 72 hours:

image

**Comment by d3athrow** [comment]
👎

**Comment by DeityLink** [comment]
Well the chicken has spoketh
